### PR TITLE
Improve filters matching and reduce unnecessary calls

### PR DESCRIPTION
**Pull request by wKich**
## Motivation

There is #188 issue with slow `getComputedStyle` call in jsdom so it makes impossible to use interactors with jest. That function is called only in one place inside the `visible` filter of `HTML` interactor.

## Approach

So my approach is to reduce `getComputedStyle` calls as much as possible. That means:
- For the `exists` method we don't actually need to find all matched elements, we can stop matching right after the first match
- Because the `visible` filter has implicitly defined default value, we need to call implicitly defined filters after explicit ones (those that defined as the second argument to interactor's call, like `Button('Submit', { focused: true })`

PS Actually the code already implements this logic from 2nd point, but I noticed that filter matchers are called immediately inside their constructor, that means with `Button('Submit', { focused: true })` even if we found other buttons with different locator or they are not focused we still check `visible` filter.

### CODE REVIEW

**Review by taras — APPROVED** [review]

**Comment by cowboyd** [comment]
These changes are a bit hard to follow since they are packaged along with an internal re-organization of the files. I do appreciate keeping things organized and having related functionality separate and add clarity to what's going on, but doing so along with the change in filter logic only serves to obscure what exactly is different before and after since you have to try and hold in your head the difference between two separate files and not blocks of lines within the same file.
